### PR TITLE
KerasNLP: Abstractive Summarization with BART

### PR DESCRIPTION
**Pull request by abheesht17**

### CODE REVIEW

File: examples/nlp/abstractive_summarization_with_bart.py
Comment (mattdangerw) [inline examples/nlp/abstractive_summarization_with_bart.py:41]: we can probably make this install line a one liner, at least after we land whatever fixes we need for keras-nlp

File: examples/nlp/abstractive_summarization_with_bart.py
Comment (mattdangerw) [inline examples/nlp/abstractive_summarization_with_bart.py:55]: we can try rewriting this to keras-core (can ditch mixed precision for now while we debug that if it helps)

File: examples/nlp/abstractive_summarization_with_bart.py
Comment (mattdangerw) [inline examples/nlp/abstractive_summarization_with_bart.py:77]: where is this drive link from? can we not use keras.utils.get_file?

File: examples/nlp/abstractive_summarization_with_bart.py
Comment (mattdangerw) [inline examples/nlp/abstractive_summarization_with_bart.py:111]: I think the prefetch call only really makes sense right before the model, e.g. after tokenization.

File: examples/nlp/abstractive_summarization_with_bart.py
Comment (mattdangerw) [inline examples/nlp/abstractive_summarization_with_bart.py:227]: Figuring out these metrics is going to be a little tricky. Since they are text manipulations we will need the tf backend (as torch and jax lack string types). What we could do is update our metrics in keras-nlp to convert any inputs with something like `tf.convert_to_tensor`, then this would work on all backends. Might need a little playing around.

Alternately, if we don't want the headache, we could just ditch this section for now.

**Review by mattdangerw — COMMENTED** [review]
This looks good! Should we try converting it to a multi-backend example?

File: examples/nlp/abstractive_summarization_with_bart.py
Comment (abheesht17) [reply examples/nlp/abstractive_summarization_with_bart.py:77]: Followed the manual download instructions here: https://www.tensorflow.org/datasets/catalog/samsum. The provided link intermittently fails with `wget` (and `keras.utils.get_file`, by extension). I uploaded it to my GDrive. `keras.utils.get_file` might not work on a GDrive link (wget wasn't working), probably because GDrive allows downloading files below a certain size

File: examples/nlp/abstractive_summarization_with_bart.py
Comment (abheesht17) [reply examples/nlp/abstractive_summarization_with_bart.py:227]: Let's ditch it for now, then. I'll open an issue on KerasNLP for porting metrics over

**Comment by abheesht17** [comment]
@mattdangerw - this is ready!

File: examples/nlp/abstractive_summarization_with_bart.py
Comment (mattdangerw) [reply examples/nlp/abstractive_summarization_with_bart.py:88]: Probably best to avoid this if we could. If the provided link intermittently fails, maybe try the huggingface location?

https://huggingface.co/datasets/samsum/blob/main/data/corpus.7z

```
filename = keras.utils.get_file(
    "corpus.7z",
    origin="https://huggingface.co/datasets/samsum/resolve/main/data/corpus.7z"
)
```

**Review by mattdangerw — APPROVED** [review]
Looks good! Just a few comments.

**Comment by abheesht17** [comment]
Ready for merge!

**Comment by mattdangerw** [comment]
All looks good to me! Note that we have not actually stuck BART in a release yet, but hopefully we can ready one with gpt-neox, pythia, bart and whisper in the next week or two.

I am fine if we just release the guide based on the master branch for now. We could also wait for the release to better promote on social.

Maybe easiest to release this as it's ready, and tweet about it again when the next KerasNLP release is out.

**Review by fchollet — APPROVED** [review]
Thank you for the contribution!

**Pull request merged**
